internal/dag/extension_processor.go

```diff
@@ -148,21 +160,16 @@ func (p *ExtensionServiceProcessor) buildExtensionService(
 
 		svc, port, err := cache.LookupService(svcName, intstr.FromInt(target.Port))
 		if err != nil {
-			// TODO(jpeach): Add status condition, #2874.
-			p.WithError(err).
-				WithField("name", svcName).
-				WithField("port", target.Port).
-				Error("failed to look up service")
-			return nil
+			validCondition.AddErrorf("ServiceError", "ServiceUnresolvedReference",
```

Comment:
Do we have any plans to make these categories lookups? Meaning `ServiceError` & `ServiceUnresolvedReference` should have some sort of central place along with docs that explain their meaning.

Reply:
Yeh we should do this. I did this change quickly because I was told it was needed for the release. I'll file a separate issue to do add contents which goods can be attached to. Not sure we need to expose these names as user docs.

Reply:
I filed https://github.com/projectcontour/contour/issues/2988